gozstd.go

```diff
@@ -268,6 +288,9 @@ func decompress(dctx, dctxDict *dctxWrapper, dst, src []byte, dd *DDict) ([]byte
 	decompressBound++
 
 	if n := dstLen + decompressBound - cap(dst); n > 0 {
+		if maxAlloc > 0 && n > maxAlloc {
```

Comment:
This check must be unconditionally applied to all the decompressed data, even if it fits the `dst` capacity. This will simplify the usage of the decompression API.